.github/workflows/CI.yml

```diff
@@ -0,0 +1,128 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - main
+
+  workflow_dispatch:
+
+jobs:
+  test:
+    runs-on: [self-hosted, "${{ matrix.archconfig }}", gcc, lite, "2204"]
+
+    strategy:
+      matrix:
+        archconfig: [x86_64, aarch64, armv7l]
+        build_type: [Debug, Release]
+      fail-fast: false
+
+    steps:
+      - name: Cleanup previous jobs
+        run: |
+          echo "Cleaning up previous runs"
+          sudo rm -rf ${{ github.workspace }}/*
+          sudo rm -rf ${{ github.workspace }}/.??*
+
+      - name: Install dependencies for tests
+        id: install_dependencies
+        run: sudo apt update && sudo apt-get install -y lcov g++ valgrind
```

Comment:
should be redundant as we've added that to the container image build process: https://github.com/nubificus/github-actions-runner/blob/b4663d1a26ecfed8a70cba2d356351a9c29d0d30/Dockerfile.base-2204#L71 

Seems we're missing lcov & valgrind, so we can keep it for now, but once we update the runners, we should remove that.